src/App.js

```diff
@@ -1,158 +0,0 @@
-import "./styles.css";
-import React from "react";
-import { useCallback } from "react";
-
-export default function App() {
-  const [classCheckbox, setClassCheckbox] = React.useState(false);
-  const [checkboxes, setCheckboxes] = React.useState([
-    {
-      name: "Section 1",
-      isParentChecked: false,
-      id: 1,
-      children: [
-        { id: 1, isChecked: false },
-        { id: 1, isChecked: false },
-        { id: 1, isChecked: false },
-      ],
-    },
-    {
-      name: "Section 2",
-      isParentChecked: false,
-      id: 2,
-      children: [
-        { id: 2, isChecked: false },
-        { id: 2, isChecked: false },
-        { id: 2, isChecked: false },
-      ],
-    },
-    {
-      name: "Section 3",
-      isParentChecked: false,
-      id: 3,
-      children: [
-        { id: 3, isChecked: false },
-        { id: 3, isChecked: false },
-        { id: 3, isChecked: false },
-      ],
-    },
-  ]);
-  const handleChange = (id) => {
-    const copy = [...checkboxes];
-    const index = checkboxes.findIndex((ele) => ele.id === id);
-    const output = checkboxes.find((ele) => ele.id === id);
-    output.isParentChecked = !output.isParentChecked;
-    output.children = output.children.map((ele) => {
-      return {
-        ...ele,
-        isChecked: !ele.isChecked,
-      };
-    });
-    copy[index] = output;
-   
-
-    setCheckboxes([...copy]);
-    handleClassChange();
-  };
-
-  const handleIndividualChange = (e, id, ind) => {
-    const copy = [...checkboxes];
-    const index = checkboxes.findIndex((ele) => ele.id === id);
-    const output = copy.find((ele) => ele.id === id);
-
-    output.children = output.children.map((child, index) => {
-      if (index === ind) {
-        return {
-          ...child,
-          isChecked: !child.isChecked,
-        };
-      } else {
-        return child;
-      }
-    });
```

Comment:
Can we directly use the index to update the state instead of iterating over the array ?